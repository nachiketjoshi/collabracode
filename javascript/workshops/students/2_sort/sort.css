
body {
  width: 100%;
  height: 100%;
}

table {
  font-size: 200%;
}

#run-sort-button {
  width: 10em;
  border-width: 2px;
  border-color: black;
  border-style: outset;
  background-color: aquamarine;
  text-align: center;
  vertical-align: center;
}

.attempt {
  margin-top: 2em;
  margin-bottom: 2em;
}

.element {
  border-width: 1px;
  border-color: black;
  border-style: outset;
  min-width: 2em;
  min-height: 2em;
  text-align: center;
  background-color: #E3E3E3;
}