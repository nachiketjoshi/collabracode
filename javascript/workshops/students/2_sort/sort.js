
function runSortOnce(iteration) {
  var selection = document.getElementById("sort_algorithm");
  if(!selection) {
    console.log("could not find selection");
    return true;
  }
  var sorttype = selection.item(selection.selectedIndex).value;
  if(!sorttype) {
    console.log("no sort type specified");
    return true;
  }
  var itersortfuncname = "itersort_" + sorttype;
  var itersortfunc = window[itersortfuncname];
  if(!itersortfunc) {
    console.log("no itersort function named ", itersortfuncname);
    return true;
  }
  var table = document.getElementById("sort-table");
  var tbody = table.tBodies[0];
  var lastRow = tbody.rows[tbody.rows.length-1];
  var cells = lastRow.cells;
  var arr = [];
  var arr2 = [];
  for(var i=0; i<cells.length; i++) {
    var value = parseInt(getInnerText(cells[i]));
    arr.push(value);
    arr2.push(value);
  }
  var maxiterations = 0;
  switch(sorttype) {
    case "bubblesort":
      maxiterations = arr.length - 1;
      break;
    case "insertionsort":
      maxiterations = arr.length - 1;
      break;
    case "radixsort":
      maxiterations = radixsort_numPasses(arr, 10);
      break;
    default:
      maxiterations = -1;
      break;
  }
  console.log("running through iteration ", iteration, " out of ", maxiterations, " for sort ", sorttype);
  if(iteration >= maxiterations) {
    console.log("no more iterations");
    return true;
  }
  var outarr = itersortfunc(arr, iteration);
  var newRow = document.createElement("tr");
  newRow.className = "attempt";
  for(var i=0; i<outarr.length; i++) {
    var newCell = document.createElement("td");
    newCell.className = "element";
    setInnerText(newCell, new String(outarr[i]));
    newRow.appendChild(newCell);
  }
  tbody.appendChild(newRow);
  if(iteration + 1 >= maxiterations) {
    console.log("no more iterations");
    return true;
  }
  return false;
}

function runSort() {
  for(var i=0; !runSortOnce(i); i++) {}
  console.log("done");
}

function attachButtonEvents() {
  addEvent(document.getElementById("run-sort-button"), "click", runSort);
}

addEvent(window, "load", attachButtonEvents);

