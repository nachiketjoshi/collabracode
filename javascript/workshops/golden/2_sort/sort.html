<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE html >
<html>
  <head>
    <title>Lesson 1 -- Basic JavaScript</title>
    <script type="text/javascript" src="util.js"></script>
    <script type="text/javascript" src="student_sort.js"></script>
    <script type="text/javascript" src="sort.js"></script>
    <link rel="stylesheet" href="sort.css"/>
  </head>
  <body>
    <h1>Sorting</h1>
    <p>
      <label for="sort_algorithm">Sorting Algorithm</label>
      <select id="sort_algorithm" name="sort_algorithm">
        <option value="bubblesort">Bubble Sort</option>
        <option value="insertionsort">Insertion Sort</option>
        <option value="radixsort">Radix Sort</option>
      </select>
    </p>
    <div id="run-sort-button">Click to Run Sort</div>
    <table id="sort-table">
      <tbody>
        <tr class="attempt">
          <td class="element">23</td>
          <td class="element">17</td>
          <td class="element">53</td>
          <td class="element">29</td>
          <td class="element">47</td>
          <td class="element">19</td>
          <td class="element">91</td>
          <td class="element">73</td>
          <td class="element">71</td>
          <td class="element">5</td>
          <td class="element">13</td>
        </tr>
      </tbody>
    </table>
    <div>
      <div>
        To make this work, open up <code>sort.js</code> and fill in the following functions:
      </div>
      <ul>
        <li><b>itersort_bubblesort</b> -- Implement algorithm specified at <a target="_blank" href="http://en.wikipedia.org/wiki/Bubble_sort">http://en.wikipedia.org/wiki/Bubble_sort</a></li>
        <li><b>itersort_insertionsort</b> -- Implement algorithm specified at <a target="_blank" href="http://en.wikipedia.org/wiki/Insertion_sort">http://en.wikipedia.org/wiki/Insertion_sort</a></li>
        <li><b>itersort_radixsort</b> -- Implement algorithm specified at <a target="_blank" href="http://en.wikipedia.org/wiki/Radix_sort">http://en.wikipedia.org/wiki/Radix_sort</a></li>
      </ul>
      <h4>Testing</h4>
      <p>
        If on Linux or Mac, go to git base directory and run <code>python -m SimpleHTTPServer</code>.<br/>
        Update <code>sort_test.html</code> by filling in the test cases.<br/>
        Go to <a target="_blank" href="http://localhost:8000/jsunit/testRunner.html">http://localhost:8000/jsunit/testRunner.html</a><br/>
        Run the test at <a target="_blank" href="http://localhost:8000/tutorials/students/2_sort/sort_test.html">http://localhost:8000/tutorials/students/2_sort/sort_test.html</a>
      </p>
    </div>
  </body>
</html>
